### ISSUE TITLE
Entries from pg_worker_list.conf aren't picked up in PostgreSQL 11.

### ISSUE THREAD

**Issue by pykello** [citusdata/citus#2453]
When using postgres 11 and master branch of citus, entries from `pg_worker_list.conf` aren't picked up properly. I can see them in `pg_dist_node`, but `master_get_active_worker_nodes()` returns empty set. This doesn't happen when using postgres 10 and master branch of citus.

I used [this script](https://gist.github.com/pykello/ec9d391b5d61b6352d47ba6f8e93c5cc) for testing.

The output when using postgres 11 is:

```
Setting up data.9700 ...
+ Removing data dir
+ postgres version is: PostgreSQL 11.0
+ initdb
+ Setting config
+ Adding pg_worker_list.conf
+ Starting postgres
+ pg_dist_node:
 nodeid | groupid | nodename  | nodeport | noderack | hasmetadata | isactive | noderole | nodecluster 
--------+---------+-----------+----------+----------+-------------+----------+----------+-------------
      1 |       1 | localhost |     9701 | default  | f           | t        | primary  | default
      2 |       2 | localhost |     9702 | default  | f           | t        | primary  | default
(2 rows)

+ master_get_active_worker_nodes()
 node_name | node_port 
-----------+-----------
(0 rows)

                                            citus_version                                            
-----------------------------------------------------------------------------------------------------
 Citus 8.0devel on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0, 64-bit
(1 row)
```

when using postgres 10, the output is as expected:

```
Setting up data.9700 ...
+ Shutting down previously running postgres
+ Removing data dir
+ postgres version is: PostgreSQL 10.5
+ initdb
+ Setting config
+ Adding pg_worker_list.conf
+ Starting postgres
+ pg_dist_node:
 nodeid | groupid | nodename  | nodeport | noderack | hasmetadata | isactive | noderole | nodecluster 
--------+---------+-----------+----------+----------+-------------+----------+----------+-------------
      1 |       1 | localhost |     9701 | default  | f           | t        | primary  | default
      2 |       2 | localhost |     9702 | default  | f           | t        | primary  | default
(2 rows)

+ master_get_active_worker_nodes()
 node_name | node_port 
-----------+-----------
 localhost |      9701
 localhost |      9702
(2 rows)

                                            citus_version                                            
-----------------------------------------------------------------------------------------------------
 Citus 8.0devel on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0, 64-bit
(1 row)
```

This breaks some of our test scripts, so I think we should fix this. It is late here right now, so if no one finds the cause by Wednesday morning, I'll look into it.

**Comment by pykello** [comment]
Note that the issue happens only when adding nodes using `pg_worker_list.conf`. Using `master_add_node()` works as expected.

**Comment by pykello** [comment]
I repeated the test in an EC2 instance to make sure that it's not because of something in my machine. I saw the same behaviour there.

**Comment by pykello** [comment]
After some investigation, the issue is because of a new performance improvement in PG11: [Fast ALTER TABLE ADD COLUMN with a non-NULL default](https://github.com/postgres/postgres/commit/16828d5c0273b4fe5f10f42588005f16b415b2d8)

Now we need to do heap_expand_tuple() before being able to access attributes added after ALTER TABLE ... ADD COLUMN while there is data in the table. In the particular case of pg_worker_list.conf, after reading the file into pg_dist_node, we added couple more columns which caused the problem.

In other cases, this can cause a problem when upgrading Citus. For example, if someone is already using Citus 7.5 on PG11, and upgrades to Citus 8.0 on PG11 without recreating the whole database, the problem can show itself if we have added columns to any of the catalog tables. This upgrade path might be rare since PG11 is new, but it will mos likely happen in the next release of Citus.

So I guess this is not urgent for this release, but is urgent for the next release, unless someone knows a case other than `pg_dist_node` that gets populated while doing the `CREATE EXTENSION citus;`.
